packages/api/src/lib/postgres/models/createModel.batch.spec.js

```diff
@@ -0,0 +1,194 @@
+                uncertaintyInMeters: 10,
+              },
+              verticalPosition: {
+                maximumDepthInMeters: 100,
+                maximumElevationInMeters: 100,
+                minimumDepthInMeters: 20,
+                minimumElevationInMeters: 20,
+              },
+            },
+            monthEnd: 1,
+            monthStart: 1,
+            occurrenceDateText: '15 jan 1986',
+            yearEnd: 1986,
+            yearStart: 1986,
+          },
+        ],
+        originStandardized: 'Standardized origin',
+        physicalUnits: [
+          {
+            alternateIdentifiersText: 'alternateIdentifiersText',
+            catalogedUnit: {
+              catalogNumber: `${catalogNumber}`,
+              publishRecord: true,
+              storedUnderTaxonName: 'Sorex minutus',
+            },
+            normalStorageLocationText: 'normalStorageLocationText',
+            physicalUnitText: 'physicalUnitText',
+          },
+        ],
+      },
+    },
+  }
+}
+
+const setup = () => {
+  return createDb({ config }).then(sequelize => {
+    const model = createModel({
+      name: 'test',
+      schemaModelName: null,
+      schemaVersion: '0',
+      sequelize,
+    })
+    return syncModels({
+      config,
+      models: {
+        test: model,
+      },
+    }).then(() => {
+      return model
+    })
+  })
+}
+
+const maxAllowedTime = 100000
+
+batchDescribe('lib/postgres/models', () => {
+  let model
+  describe('createModel batch', () => {
+    beforeAll(() => {
+      return setup().then(createdModel => {
+        model = createdModel
+      })
+    })
+
+    describe('create', () => {
+      it(
+        'Creates and returns a simple record',
+        () => {
+          const createEntry = index => {
+            return {
+              document: createData(index),
+            }
+          }
+
+          const execute = batchData => {
+            return model.Model.bulkCreate(batchData).then(res => {
+              return res
+            })
+          }
+
+          return batchExecute({
+            createEntry,
+            execute,
+            numberOfEntries: 300000,
+            numberOfentriesEachBatch: 10000,
+          }).then(() => {
+            expect(1).toBe(1)
+          })
+        },
+        maxAllowedTime
+      )
+    })
+  })
+})
+
+// 100k
+// 10k each batch
+// 90000.batch.execute: 2875.403ms
+// console.log console.js:171
+//   batch: 28401.806ms
+
+// 100k
+// 1k each batch
+//   99000.batch.execute: 970.829ms
+// console.log console.js:171
+//   batch: 32011.944ms
+// console.log src/lib/postgres/models/createModel.batch.spec.js:165
+//   test:batch
+
+// 300k
+// 1k each batch
+// console.log console.js:171
+//   290000.batch.execute: 3241.546ms
+// console.log console.js:171
+//   batch: 85810.160ms
+// console.log src/lib/postgres/models/createModel.batch.spec.js:165
+//   test:batch
+
+// 100k
+// 100k each batch - heap out of memory
```

Comment:
want to keep those? maybe use `/* ... */` syntax and add a comment then

Reply:
I leave it like this one but will get back to it